File: OCDemo_v6.5.xx/iOSDemo/Others/Widget/DemoADFootView.m
//
//  DemoADFootView.m
//  AnyThingSDKDemo
//
//  Created by ltz on 2021/12/6.
//

#import "DemoADFootView.h"
#import "Masonry.h"

@interface DemoADFootView()
@property (nonatomic, assign) BOOL isNeetRemove;
@property (nonatomic, assign) BOOL isNeetHidenAndMove;
@end

@implementation DemoADFootView

- (instancetype)initWithRemoveBtn
{
    if (self = [super init]) {
        [self setupUIWithRemovebtn:YES hidenBtn:NO];
    }
    return self;
}

- (instancetype)initWithRemoveAndHidenBtn
{
    if (self = [super init]) {
        [self setupUIWithRemovebtn:YES hidenBtn:YES];
    }
    return self;
}

- (instancetype)init
{
    if (self = [super init]) {
        [self setupUIWithRemovebtn:NO hidenBtn:NO];
    }
    return self;
}

- (void)setupUIWithRemovebtn:(BOOL)haveRemove hidenBtn:(BOOL)haveHiden
{
    self.isNeetRemove = haveRemove;
    self.isNeetHidenAndMove = haveHiden;

    [self addSubview:self.loadBtn];
    [self addSubview:self.showBtn];
    [self addSubview:self.logBtn];
    if (haveRemove) {
        [self addSubview:self.removeBtn];
    }
    if (haveHiden) {
        [self addSubview:self.hidenBtn];
        [self addSubview:self.reShowBtn];
    }
}

- (void)layoutSubviews
{
    if (self.isNeetRemove) {
        // 需要remove按钮
        if (self.isNeetHidenAndMove) {
            // 需要hiden和Move按钮
            [self.loadBtn mas_updateConstraints:^(MASConstraintMaker *make) {
                make.left.equalTo(self.mas_left).offset(kAdaptW(26,26));
                make.width.mas_equalTo((kScreenW - kAdaptW(26 * 3,26 * 3)) / 2);
                make.height.mas_equalTo(kAdaptW(90,90));
                make.top.equalTo(self.mas_top).offset(kAdaptW(10,10));
            }];
            
            [self.showBtn mas_updateConstraints:^(MASConstraintMaker *make) {
                make.left.equalTo(self.mas_left).offset(kAdaptW(26,26));
                make.width.mas_equalTo((kScreenW - kAdaptW(26 * 3,26 * 3)) / 2);
                make.height.mas_equalTo(kAdaptW(90,90));
                make.top.equalTo(self.loadBtn.mas_bottom).offset(kAdaptW(20,20));
            }];
            
            [self.logBtn mas_updateConstraints:^(MASConstraintMaker *make) {
                make.left.equalTo(self.mas_left).offset(kAdaptW(26,26));
                make.width.mas_equalTo((kScreenW - kAdaptW(26 * 3,26 * 3)) / 2);
                make.height.mas_equalTo(kAdaptW(90,90));
                make.top.equalTo(self.showBtn.mas_bottom).offset(kAdaptW(20,20));
            }];
            
            [self.removeBtn mas_updateConstraints:^(MASConstraintMaker *make) {
                make.left.equalTo(self.loadBtn.mas_right).offset(kAdaptW(26,26));
                make.right.equalTo(self.mas_right).offset(kAdaptW(-26,-26));
                make.height.mas_equalTo(kAdaptW(90,90));
                make.top.equalTo(self.loadBtn.mas_top);
            }];
            
            [self.hidenBtn mas_updateConstraints:^(MASConstraintMaker *make) {
                make.left.equalTo(self.showBtn.mas_right).offset(kAdaptW(26,26));
                make.right.equalTo(self.mas_right).offset(kAdaptW(-26,-26));
                make.height.mas_equalTo(kAdaptW(90,90));
                make.top.equalTo(self.logBtn.mas_top);
            }];
             
            [self.reShowBtn mas_updateConstraints:^(MASConstraintMaker *make) {
                make.left.equalTo(self.logBtn.mas_right).offset(kAdaptW(26,26));
                make.right.equalTo(self.mas_right).offset(kAdaptW(-26,-26));
                make.height.mas_equalTo(kAdaptW(90,90));
                make.top.equalTo(self.showBtn.mas_top);
            }];
            
        } else {
            [self.loadBtn mas_updateConstraints:^(MASConstraintMaker *make) {
                make.left.equalTo(self.mas_left).offset(kAdaptW(26,26));
                make.right.equalTo(self.mas_right).offset(kAdaptW(-26,-26));
                make.height.mas_equalTo(kAdaptW(90,90));
                make.top.equalTo(self.mas_top).offset(kAdaptW(10,10));
            }];
            
            [self.showBtn mas_updateConstraints:^(MASConstraintMaker *make) {
                make.left.equalTo(self.mas_left).offset(kAdaptW(26,26));
                make.right.equalTo(self.mas_right).offset(kAdaptW(-26,-26));
                make.height.mas_equalTo(kAdaptW(90,90));
                make.top.equalTo(self.loadBtn.mas_bottom).offset(kAdaptW(20,20));
            }];
            
            [self.logBtn mas_updateConstraints:^(MASConstraintMaker *make) {
                make.left.equalTo(self.mas_left).offset(kAdaptW(26,26));
                make.width.mas_equalTo((kScreenW - kAdaptW(26 * 3,26 * 3)) / 2);
                make.height.mas_equalTo(kAdaptW(90,90));
                make.top.equalTo(self.showBtn.mas_bottom).offset(kAdaptW(20,20));
            }];
            
            [self.removeBtn mas_updateConstraints:^(MASConstraintMaker *make) {
                make.left.equalTo(self.logBtn.mas_right).offset(kAdaptW(26,26));
                make.right.equalTo(self.mas_right).offset(kAdaptW(-26,-26));
                make.height.mas_equalTo(kAdaptW(90,90));
                make.top.equalTo(self.logBtn.mas_top);
            }];
        }
    } else {
        [self.loadBtn mas_updateConstraints:^(MASConstraintMaker *make) {
            make.left.equalTo(self.mas_left).offset(kAdaptW(26,26));
            make.right.equalTo(self.mas_right).offset(kAdaptW(-26,-26));
            make.height.mas_equalTo(kAdaptW(90,90));
            make.top.equalTo(self.mas_top).offset(kAdaptW(10,10));
        }];
        
        [self.showBtn mas_updateConstraints:^(MASConstraintMaker *make) {
            make.left.equalTo(self.mas_left).offset(kAdaptW(26,26));
            make.right.equalTo(self.mas_right).offset(kAdaptW(-26,-26));
            make.height.mas_equalTo(kAdaptW(90,90));
            make.top.equalTo(self.loadBtn.mas_bottom).offset(kAdaptW(20,20));
        }];
        
        [self.logBtn mas_updateConstraints:^(MASConstraintMaker *make) {
            make.left.equalTo(self.mas_left).offset(kAdaptW(26,26));
            make.right.equalTo(self.mas_right).offset(kAdaptW(-26,-26));
            make.height.mas_equalTo(kAdaptW(90,90));
            make.top.equalTo(self.showBtn.mas_bottom).offset(kAdaptW(20,20));
        }];
    }
}

#pragma mark - Action
- (void)clickLoadBtn
{
    if (self.clickLoadBlock) {
        self.clickLoadBlock();
    }
}

- (void)clickLogBtn
{
    if (self.clickLogBlock) {
        self.clickLogBlock();
    }
}
 
- (void)clickShowBtn
{
    if (self.clickShowBlock) {
        self.clickShowBlock();
    }
}

- (void)clickRemoveBtn
{
    if (self.clickRemoveBlock) {
        self.clickRemoveBlock();
    }
}

- (void)clickReShowBtn
{
    if (self.clickReShowBlock) {
        self.clickReShowBlock();
    }
}

- (void)clickHidenBtn
{
    if (self.clickHidenBlock) {
        self.clickHidenBlock();
    }
}
 
- (UIImage *)imageWithColor:(UIColor *)color {
    CGRect rect = CGRectMake(0.0f, 0.0f, 1.0f, 1.0f);
    UIGraphicsBeginImageContext(rect.size);
    CGContextRef context = UIGraphicsGetCurrentContext();
    CGContextSetFillColorWithColor(context, [color CGColor]);
    CGContextFillRect(context, rect);
    UIImage *image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    return image;
}

#pragma mark - lazy
- (UIButton *)loadBtn
{
    if (!_loadBtn) {
        _loadBtn = [[UIButton alloc] init];
        _loadBtn.layer.masksToBounds = YES;
        _loadBtn.layer.cornerRadius = 5;
        [_loadBtn setTitle:kLocalizeStr(@"加载广告") forState:UIControlStateNormal];
        _loadBtn.layer.borderColor = kRGB(73, 109, 255).CGColor;
        _loadBtn.layer.borderWidth = kAdaptW(3,3);
        [_loadBtn setTitleColor:[UIColor whiteColor] forState:UIControlStateHighlighted];
        [_loadBtn setTitleColor:kRGB(73, 109, 255) forState:UIControlStateNormal];
        [_loadBtn setBackgroundImage:[self imageWithColor:kRGB(73, 109, 255)] forState:UIControlStateHighlighted];
        [_loadBtn setBackgroundImage:[self imageWithColor:[UIColor whiteColor]] forState:UIControlStateNormal];
        [_loadBtn addTarget:self action:@selector(clickLoadBtn) forControlEvents:UIControlEventTouchUpInside];
           
    }
    return _loadBtn;
}

- (UIButton *)showBtn
{
    if (!_showBtn) {
        _showBtn = [[UIButton alloc] init];
        _showBtn.layer.masksToBounds = YES;
        _showBtn.layer.cornerRadius = 5;
        [_showBtn setTitle:kLocalizeStr(@"展示广告") forState:UIControlStateNormal];
        _showBtn.layer.borderColor = kRGB(73, 109, 255).CGColor;
        _showBtn.layer.borderWidth = kAdaptW(3,3);
        [_showBtn setTitleColor:[UIColor whiteColor] forState:UIControlStateHighlighted];
        [_showBtn setTitleColor:kRGB(73, 109, 255) forState:UIControlStateNormal];
        [_showBtn setBackgroundImage:[self imageWithColor:kRGB(73, 109, 255)] forState:UIControlStateHighlighted];
        [_showBtn setBackgroundImage:[self imageWithColor:[UIColor whiteColor]] forState:UIControlStateNormal];
        [_showBtn addTarget:self action:@selector(clickShowBtn) forControlEvents:UIControlEventTouchUpInside];
    }
    return _showBtn;
}

- (UIButton *)logBtn
{
    if (!_logBtn) {
        _logBtn = [[UIButton alloc] init];
        _logBtn.layer.masksToBounds = YES;
        _logBtn.layer.cornerRadius = 5;
        [_logBtn setTitle:kLocalizeStr(@"清除日志") forState:UIControlStateNormal];
        _logBtn.layer.borderColor = kRGB(73, 109, 255).CGColor;
        _logBtn.layer.borderWidth = kAdaptW(3,3);
        [_logBtn setTitleColor:[UIColor whiteColor] forState:UIControlStateHighlighted];
        [_logBtn setTitleColor:kRGB(73, 109, 255) forState:UIControlStateNormal];
        [_logBtn setBackgroundImage:[self imageWithColor:kRGB(73, 109, 255)] forState:UIControlStateHighlighted];
        [_logBtn setBackgroundImage:[self imageWithColor:[UIColor whiteColor]] forState:UIControlStateNormal];
        [_logBtn addTarget:self action:@selector(clickLogBtn) forControlEvents:UIControlEventTouchUpInside];
    }
    return _logBtn;
}

- (UIButton *)removeBtn
{
    if (!_removeBtn) {
        _removeBtn = [[UIButton alloc] init];
        _removeBtn.layer.masksToBounds = YES;
        _removeBtn.layer.cornerRadius = 5;
        [_removeBtn setTitle:kLocalizeStr(@"移除广告") forState:UIControlStateNormal];
        _removeBtn.layer.borderColor = kRGB(73, 109, 255).CGColor;
        _removeBtn.layer.borderWidth = kAdaptW(3,3);
        [_removeBtn setTitleColor:[UIColor whiteColor] forState:UIControlStateHighlighted];
        [_removeBtn setTitleColor:kRGB(73, 109, 255) forState:UIControlStateNormal];
        [_removeBtn setBackgroundImage:[self imageWithColor:kRGB(73, 109, 255)] forState:UIControlStateHighlighted];
        [_removeBtn setBackgroundImage:[self imageWithColor:[UIColor whiteColor]] forState:UIControlStateNormal];
        
        [_removeBtn addTarget:self action:@selector(clickRemoveBtn) forControlEvents:UIControlEventTouchUpInside];
    }
    return _removeBtn;
}

- (UIButton *)reShowBtn
{
    if (!_reShowBtn) {
        _reShowBtn = [[UIButton alloc] init];
        _reShowBtn.layer.masksToBounds = YES;
        _reShowBtn.layer.cornerRadius = 5;
        [_reShowBtn setTitle:kLocalizeStr(@"重新展示") forState:UIControlStateNormal];
        _reShowBtn.layer.borderColor = kRGB(73, 109, 255).CGColor;
        _reShowBtn.layer.borderWidth = kAdaptW(3,3);
        [_reShowBtn setTitleColor:[UIColor whiteColor] forState:UIControlStateHighlighted];
        [_reShowBtn setTitleColor:kRGB(73, 109, 255) forState:UIControlStateNormal];
        [_reShowBtn setBackgroundImage:[self imageWithColor:kRGB(73, 109, 255)] forState:UIControlStateHighlighted];
        [_reShowBtn setBackgroundImage:[self imageWithColor:[UIColor whiteColor]] forState:UIControlStateNormal];
        
        [_reShowBtn addTarget:self action:@selector(clickReShowBtn) forControlEvents:UIControlEventTouchUpInside];
    }
    return _reShowBtn;
}

- (UIButton *)hidenBtn
{
    if (!_hidenBtn) {
        _hidenBtn = [[UIButton alloc] init];
        _hidenBtn.layer.masksToBounds = YES;
        _hidenBtn.layer.cornerRadius = 5;
        [_hidenBtn setTitle:kLocalizeStr(@"隐藏广告") forState:UIControlStateNormal];
        _hidenBtn.layer.borderColor = kRGB(73, 109, 255).CGColor;
        _hidenBtn.layer.borderWidth = kAdaptW(3,3);
        [_hidenBtn setTitleColor:[UIColor whiteColor] forState:UIControlStateHighlighted];
        [_hidenBtn setTitleColor:kRGB(73, 109, 255) forState:UIControlStateNormal];
        [_hidenBtn setBackgroundImage:[self imageWithColor:kRGB(73, 109, 255)] forState:UIControlStateHighlighted];
        [_hidenBtn setBackgroundImage:[self imageWithColor:[UIColor whiteColor]] forState:UIControlStateNormal];
        
        [_hidenBtn addTarget:self action:@selector(clickHidenBtn) forControlEvents:UIControlEventTouchUpInside];
    }
    return _hidenBtn;
}
@end

File: SwiftDemo/SwiftDemo/Others/Widget/DemoADFootView.swift
//
//  DemoADFootView.swift
//  SwiftDemo
//
//  Created by ltz on 2025/8/20.
//

import UIKit
import SnapKit

class DemoADFootView: UIView {
    
    // MARK: - Closures
    var clickLoadBlock: (() -> Void)?
    var clickLogBlock: (() -> Void)?
    var clickShowBlock: (() -> Void)?
    var clickRemoveBlock: (() -> Void)?
    var clickReShowBlock: (() -> Void)?
    var clickHidenBlock: (() -> Void)?
    var clickLeftMoveBlock: (() -> Void)?
    var clickRightMoveBlock: (() -> Void)?
    
    // MARK: - UI Properties
//    var loadBtn: UIButton!
//    var showBtn: UIButton!
//    var logBtn: UIButton!
//    var removeBtn: UIButton!
//    var reShowBtn: UIButton!
//    var hidenBtn: UIButton!
    
    // MARK: - Private Properties
    private var isNeetRemove: Bool = false
    private var isNeetHidenAndMove: Bool = false
    
    // MARK: - Initializers
    convenience init(withRemoveBtn: Bool) {
        self.init()
        setupUI(withRemoveBtn: true, hidenBtn: false)
    }
    
    convenience init(withRemoveAndHidenBtn: Bool) {
        self.init()
        setupUI(withRemoveBtn: true, hidenBtn: true)
    }
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        setupUI(withRemoveBtn: false, hidenBtn: false)
    }
    
    required init?(coder: NSCoder) {
        super.init(coder: coder)
        setupUI(withRemoveBtn: false, hidenBtn: false)
    }
    
    // MARK: - Setup Methods
    private func setupUI(withRemoveBtn haveRemove: Bool, hidenBtn haveHiden: Bool) {
        self.isNeetRemove = haveRemove
        self.isNeetHidenAndMove = haveHiden
        
        addSubview(loadBtn)
        addSubview(showBtn)
        addSubview(logBtn)
        
        if haveRemove {
            addSubview(removeBtn)
        }
        
        if haveHiden {
            addSubview(hidenBtn)
            addSubview(reShowBtn)
        }
    }
    
    override func layoutSubviews() {
        super.layoutSubviews()
        
        if isNeetRemove {
            // 需要remove按钮
            if isNeetHidenAndMove {
                // 需要hiden和Move按钮
                loadBtn.snp.updateConstraints { make in
                    make.left.equalTo(self.snp.left).offset(kAdaptW(26, 26))
                    make.width.equalTo((kScreenW - kAdaptW(26 * 3, 26 * 3)) / 2)
                    make.height.equalTo(kAdaptW(90, 90))
                    make.top.equalTo(self.snp.top).offset(kAdaptW(10, 10))
                }
                
                showBtn.snp.updateConstraints { make in
                    make.left.equalTo(self.snp.left).offset(kAdaptW(26, 26))
                    make.width.equalTo((kScreenW - kAdaptW(26 * 3, 26 * 3)) / 2)
                    make.height.equalTo(kAdaptW(90, 90))
                    make.top.equalTo(loadBtn.snp.bottom).offset(kAdaptW(20, 20))
                }
                
                logBtn.snp.updateConstraints { make in
                    make.left.equalTo(self.snp.left).offset(kAdaptW(26, 26))
                    make.width.equalTo((kScreenW - kAdaptW(26 * 3, 26 * 3)) / 2)
                    make.height.equalTo(kAdaptW(90, 90))
                    make.top.equalTo(showBtn.snp.bottom).offset(kAdaptW(20, 20))
                }
                
                removeBtn.snp.updateConstraints { make in
                    make.left.equalTo(loadBtn.snp.right).offset(kAdaptW(26, 26))
                    make.right.equalTo(self.snp.right).offset(kAdaptW(-26, -26))
                    make.height.equalTo(kAdaptW(90, 90))
                    make.top.equalTo(loadBtn.snp.top)
                }
                
                hidenBtn.snp.updateConstraints { make in
                    make.left.equalTo(showBtn.snp.right).offset(kAdaptW(26, 26))
                    make.right.equalTo(self.snp.right).offset(kAdaptW(-26, -26))
                    make.height.equalTo(kAdaptW(90, 90))
                    make.top.equalTo(logBtn.snp.top)
                }
                
                reShowBtn.snp.updateConstraints { make in
                    make.left.equalTo(logBtn.snp.right).offset(kAdaptW(26, 26))
                    make.right.equalTo(self.snp.right).offset(kAdaptW(-26, -26))
                    make.height.equalTo(kAdaptW(90, 90))
                    make.top.equalTo(showBtn.snp.top)
                }
                
            } else {
                loadBtn.snp.updateConstraints { make in
                    make.left.equalTo(self.snp.left).offset(kAdaptW(26, 26))
                    make.right.equalTo(self.snp.right).offset(kAdaptW(-26, -26))
                    make.height.equalTo(kAdaptW(90, 90))
                    make.top.equalTo(self.snp.top).offset(kAdaptW(10, 10))
                }
                
                showBtn.snp.updateConstraints { make in
                    make.left.equalTo(self.snp.left).offset(kAdaptW(26, 26))
                    make.right.equalTo(self.snp.right).offset(kAdaptW(-26, -26))
                    make.height.equalTo(kAdaptW(90, 90))
                    make.top.equalTo(loadBtn.snp.bottom).offset(kAdaptW(20, 20))
                }
                
                logBtn.snp.updateConstraints { make in
                    make.left.equalTo(self.snp.left).offset(kAdaptW(26, 26))
                    make.width.equalTo((kScreenW - kAdaptW(26 * 3, 26 * 3)) / 2)
                    make.height.equalTo(kAdaptW(90, 90))
                    make.top.equalTo(showBtn.snp.bottom).offset(kAdaptW(20, 20))
                }
                
                removeBtn.snp.updateConstraints { make in
                    make.left.equalTo(logBtn.snp.right).offset(kAdaptW(26, 26))
                    make.right.equalTo(self.snp.right).offset(kAdaptW(-26, -26))
                    make.height.equalTo(kAdaptW(90, 90))
                    make.top.equalTo(logBtn.snp.top)
                }
            }
        } else {
            loadBtn.snp.updateConstraints { make in
                make.left.equalTo(self.snp.left).offset(kAdaptW(26, 26))
                make.right.equalTo(self.snp.right).offset(kAdaptW(-26, -26))
                make.height.equalTo(kAdaptW(90, 90))
                make.top.equalTo(self.snp.top).offset(kAdaptW(10, 10))
            }
            
            showBtn.snp.updateConstraints { make in
                make.left.equalTo(self.snp.left).offset(kAdaptW(26, 26))
                make.right.equalTo(self.snp.right).offset(kAdaptW(-26, -26))
                make.height.equalTo(kAdaptW(90, 90))
                make.top.equalTo(loadBtn.snp.bottom).offset(kAdaptW(20, 20))
            }
            
            logBtn.snp.updateConstraints { make in
                make.left.equalTo(self.snp.left).offset(kAdaptW(26, 26))
                make.right.equalTo(self.snp.right).offset(kAdaptW(-26, -26))
                make.height.equalTo(kAdaptW(90, 90))
                make.top.equalTo(showBtn.snp.bottom).offset(kAdaptW(20, 20))
            }
        }
    }
    
    // MARK: - Action Methods
    @objc private func clickLoadBtn() {
        clickLoadBlock?()
    }
    
    @objc private func clickLogBtn() {
        clickLogBlock?()
    }
    
    @objc private func clickShowBtn() {
        clickShowBlock?()
    }
    
    @objc private func clickRemoveBtn() {
        clickRemoveBlock?()
    }
    
    @objc private func clickReShowBtn() {
        clickReShowBlock?()
    }
    
    @objc private func clickHidenBtn() {
        clickHidenBlock?()
    }
    
    // MARK: - Helper Methods
    private func imageWithColor(_ color: UIColor) -> UIImage {
        let rect = CGRect(x: 0.0, y: 0.0, width: 1.0, height: 1.0)
        UIGraphicsBeginImageContext(rect.size)
        let context = UIGraphicsGetCurrentContext()
        context?.setFillColor(color.cgColor)
        context?.fill(rect)
        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()
        return image ?? UIImage()
    }
    
    // MARK: - Lazy Properties
    lazy var loadBtn: UIButton = {
        let button = UIButton()
        button.layer.masksToBounds = true
        button.layer.cornerRadius = 5
        button.setTitle(kLocalizeStr("Load Ad"), for: .normal)
        button.layer.borderColor = kRGB(73, 109, 255).cgColor
        button.layer.borderWidth = kAdaptW(3, 3)
        button.setTitleColor(.white, for: .highlighted)
        button.setTitleColor(kRGB(73, 109, 255), for: .normal)
        button.setBackgroundImage(imageWithColor(kRGB(73, 109, 255)), for: .highlighted)
        button.setBackgroundImage(imageWithColor(.white), for: .normal)
        button.addTarget(self, action: #selector(clickLoadBtn), for: .touchUpInside)
        return button
    }()
    
    lazy var showBtn: UIButton = {
        let button = UIButton()
        button.layer.masksToBounds = true
        button.layer.cornerRadius = 5
        button.setTitle(kLocalizeStr("Show Ad"), for: .normal)
        button.layer.borderColor = kRGB(73, 109, 255).cgColor
        button.layer.borderWidth = kAdaptW(3, 3)
        button.setTitleColor(.white, for: .highlighted)
        button.setTitleColor(kRGB(73, 109, 255), for: .normal)
        button.setBackgroundImage(imageWithColor(kRGB(73, 109, 255)), for: .highlighted)
        button.setBackgroundImage(imageWithColor(.white), for: .normal)
        button.addTarget(self, action: #selector(clickShowBtn), for: .touchUpInside)
        return button
    }()
    
    lazy var logBtn: UIButton = {
        let button = UIButton()
        button.layer.masksToBounds = true
        button.layer.cornerRadius = 5
        button.setTitle(kLocalizeStr("Clear Log"), for: .normal)
        button.layer.borderColor = kRGB(73, 109, 255).cgColor
        button.layer.borderWidth = kAdaptW(3, 3)
        button.setTitleColor(.white, for: .highlighted)
        button.setTitleColor(kRGB(73, 109, 255), for: .normal)
        button.setBackgroundImage(imageWithColor(kRGB(73, 109, 255)), for: .highlighted)
        button.setBackgroundImage(imageWithColor(.white), for: .normal)
        button.addTarget(self, action: #selector(clickLogBtn), for: .touchUpInside)
        return button
    }()
    
    lazy var removeBtn: UIButton = {
        let button = UIButton()
        button.layer.masksToBounds = true
        button.layer.cornerRadius = 5
        button.setTitle(kLocalizeStr("Remove Ad"), for: .normal)
        button.layer.borderColor = kRGB(73, 109, 255).cgColor
        button.layer.borderWidth = kAdaptW(3, 3)
        button.setTitleColor(.white, for: .highlighted)
        button.setTitleColor(kRGB(73, 109, 255), for: .normal)
        button.setBackgroundImage(imageWithColor(kRGB(73, 109, 255)), for: .highlighted)
        button.setBackgroundImage(imageWithColor(.white), for: .normal)
        button.addTarget(self, action: #selector(clickRemoveBtn), for: .touchUpInside)
        return button
    }()
    
    lazy var reShowBtn: UIButton = {
        let button = UIButton()
        button.layer.masksToBounds = true
        button.layer.cornerRadius = 5
        button.setTitle(kLocalizeStr("Re-show"), for: .normal)
        button.layer.borderColor = kRGB(73, 109, 255).cgColor
        button.layer.borderWidth = kAdaptW(3, 3)
        button.setTitleColor(.white, for: .highlighted)
        button.setTitleColor(kRGB(73, 109, 255), for: .normal)
        button.setBackgroundImage(imageWithColor(kRGB(73, 109, 255)), for: .highlighted)
        button.setBackgroundImage(imageWithColor(.white), for: .normal)
        button.addTarget(self, action: #selector(clickReShowBtn), for: .touchUpInside)
        return button
    }()
    
    lazy var hidenBtn: UIButton = {
        let button = UIButton()
        button.layer.masksToBounds = true
        button.layer.cornerRadius = 5
        button.setTitle(kLocalizeStr("Hide Ad"), for: .normal)
        button.layer.borderColor = kRGB(73, 109, 255).cgColor
        button.layer.borderWidth = kAdaptW(3, 3)
        button.setTitleColor(.white, for: .highlighted)
        button.setTitleColor(kRGB(73, 109, 255), for: .normal)
        button.setBackgroundImage(imageWithColor(kRGB(73, 109, 255)), for: .highlighted)
        button.setBackgroundImage(imageWithColor(.white), for: .normal)
        button.addTarget(self, action: #selector(clickHidenBtn), for: .touchUpInside)
        return button
    }()
}
